lib/Runtime/Library/TypedArray.cpp

```diff
@@ -510,15 +509,12 @@ namespace Js
             if (args.Info.Count > 3 && !JavascriptOperators::IsUndefinedObject(args[3]))
             {
                 mappedLength = ArrayBuffer::ToIndex(args[3], JSERR_InvalidTypedArrayLength, scriptContext, ArrayBuffer::MaxArrayBufferLength / elementSize, false);
-                newByteLength = mappedLength * elementSize;
 
-                if (offset + newByteLength > byteLength)
+                if ((uint32)mappedLength > (byteLength - offset)/ elementSize)
```

Comment:
In the above ToIndex call you could pass (byteLength - offset)/ elementSize instead of ArrayBuffer::MaxArrayBufferLength / elementSize. That should be same, right?